### PR TITLE
Use select_for_update for ORM calls.

### LINKED ISSUE #14: Use select_for_update for updating task and worker states
Currently we're using Django's automatic transaction management when we call `get_or_create` or `update_or_create`. We should in addition use [`select_for_update`](https://docs.djangoproject.com/en/1.11/ref/models/querysets/#select-for-update) to make sure that the data we're writing to the TaskState and WorkerState models work as intended.

This is especially true if the Camera is deployed in a way that would have multiple instances run side by side for scalability reasons.

### PR DESCRIPTION
**Pull request by jezdez**
Fixes issue #14.

### CODE REVIEW

File: django_celery_monitor/managers.py [hunk 1]
```diff
@@ -1,27 +1,86 @@
...
 from django.db import connections, models, router, transaction
 
-from .utils import now
+from .utils import Now
 
 
-class TaskStateManager(models.Manager):
-    """A custom models manager for the TaskState model with some helpers."""
+class ExtendedQuerySet(models.QuerySet):
+    """A custom model manager that implements a few helpful methods."""
```
Comment (MarkusH) [inline django_celery_monitor/managers.py]: "... model _queryset_ ..."?

File: django_celery_monitor/managers.py [hunk 2]
```diff
@@ -39,3 +92,23 @@ def purge(self):
                 'DELETE FROM {0.db_table} WHERE hidden=%s'.format(meta),
```
Comment (MarkusH) [inline django_celery_monitor/managers.py]: I'm confused why you use raw sql here? What's wrong with `self.model.objects.using(router.db_for_write(self.model)).filter(hidden=True).delete()`?

File: django_celery_monitor/managers.py [hunk 3]
```diff
@@ -39,3 +92,23 @@ def purge(self):
...
+            else:
+                if states.state(state) < states.state(obj.state):
+                    keep = Task.merge_rules[states.RECEIVED]
+                    defaults = dict(
+                        (k, v) for k, v in defaults.items()
+                        if k not in keep
+                    )
+                for k, v in defaults.items():
+                    setattr(obj, k, v)
+                obj.save()
```
Comment (MarkusH) [inline django_celery_monitor/managers.py]: Care to use `update_fields=tuple(defaults)`? Also, how about this for the else branch in line 104:
```python
if states.state(state) < states.state(obj.state):
    keep = Task.merge_rules[states.RECEIVED]
else:
    keep = {}
for k, v in defaults.items():
    if k not in keep:
        setattr(obj, k, v)
obj.save(update_fields=tuple(defaults.keys()))
```

File: django_celery_monitor/managers.py (hunk 2 above)
Comment (jezdez) [inline django_celery_monitor/managers.py]: I was, too. This is purely for historical reasons, coming over from django-celery. Will fix.

File: django_celery_monitor/managers.py (hunk 3 above)
Comment (jezdez) [inline django_celery_monitor/managers.py]: Yeah, sounds good to me.

**Pull request merged**